apps/prairielearn/src/pages/authLogin/authLogin.html.ts

```diff
@@ -254,55 +257,68 @@ export function AuthLogin({
   }).toString();
 }
 
-export function AuthLoginUnsupportedProvider({
+function isLtiProvider(provider: InstitutionSupportedProvider) {
+  return provider.name === 'LTI' || provider.name === 'LTI 1.3';
+}
+
+export function AuthLoginInstitution({
   showUnsupportedMessage,
   supportedProviders,
   institutionId,
   service,
   resLocals,
 }: {
   showUnsupportedMessage: boolean;
-  supportedProviders: InstitutionSupportedProviders[];
+  supportedProviders: InstitutionSupportedProvider[];
   institutionId: string;
   service: string | null;
   resLocals: Record<string, any>;
 }) {
-  const supportsLti = supportedProviders.some((p) => p.name === 'LTI');
-  const supportsNonLti = supportedProviders.some((p) => p.name !== 'LTI');
-
-  const supportsSaml = supportedProviders.some((p) => p.name === 'SAML');
-  const supportsShib = supportedProviders.some((p) => p.name === 'Shibboleth');
-  const supportsGoogle = supportedProviders.some((p) => p.name === 'Google');
-  const supportsAzure = supportedProviders.some((p) => p.name === 'Azure');
-
-  const defaultProvider = supportedProviders.find((p) => p.is_default === true);
-  const hasNonDefaultProviders = supportedProviders.find(
-    (p) => p.name !== 'LTI' && p.is_default === false,
-  );
+  // We need to filter `supportedProviders` to reflect which ones are actually
+  // enabled in the application configuration.
+  const loginOptions = supportedProviders.filter((p) => {
+    // LTI options are always excluded from this list. An LTI session must always
+    // start from the LTI platform, not from PrairieLearn.
+    if (isLtiProvider(p)) return false;
+
+    // Some options require specific configuration to be enabled.
+    if (p.name === 'Google') return config.hasOauth;
+    if (p.name === 'Azure') return config.hasAzure && isEnterprise();
+    if (p.name === 'Shibboleth') return config.hasShib && !config.hideShibLogin;
+
+    // Assume other providers are always enabled.
+    return true;
+  });
+
+  // LTI providers were filtered out above; we need to check the original list.
+  const supportsAnyLti = supportedProviders.some((p) => isLtiProvider(p));
+  const supportsNonLti = supportedProviders.some((p) => !isLtiProvider(p));
+
+  const supportsSaml = loginOptions.some((p) => p.name === 'SAML');
+  const supportsShib = loginOptions.some((p) => p.name === 'Shibboleth');
+  const supportsGoogle = loginOptions.some((p) => p.name === 'Google');
+  const supportsAzure = loginOptions.some((p) => p.name === 'Azure');
+
+  const defaultProvider = loginOptions.find((p) => p.is_default === true);
+  const hasNonDefaultProviders = loginOptions.find((p) => p.is_default === false);
 
```

Comment:
Nit: I agree wtih @reteps 's suggestion [here](https://github.com/PrairieLearn/PrairieLearn/pull/12475#pullrequestreview-3056742277): 

```suggestion
// These variables control if providers other than the default are shown. 
```